ieproxy_darwin.go

```diff
@@ -30,79 +24,101 @@ func reloadConf() ProxyConf {
+		  ExceptionsList : <array> {
+		    0 : *.local
+		    1 : 169.254/16
+		  }
+		  FTPPassive : 1
+		  HTTPEnable : 1
+		  HTTPPort : 8081
+		  HTTPProxy : example.jp
+		  HTTPSEnable : 1
+		  HTTPSPort : 8080
+		  HTTPSProxy : example.com
+		  HTTPSUser : foo
+		  ProxyAutoConfigEnable : 1
+		  ProxyAutoConfigURLString : example.com/foo.pac
+		}
+	*/
+	inExceptionsList := false
+	exceptionsList := make([]string, 0)
+	proxyMap := map[string]string{}
+	scanner := bufio.NewScanner(bytes.NewReader(b))
+	for scanner.Scan() {
+		t := scanner.Text()
+		t = strings.TrimSpace(t)
+		if inExceptionsList {
+			if strings.Index(t, ":") > 0 {
+				s := strings.SplitN(t, ":", 2)
+				exceptionsList = append(exceptionsList, strings.TrimSpace(s[1]))
+			}
+			if t == "}" {
+				inExceptionsList = false
+			}
+			continue
+		}
+		if strings.Index(t, ":") > 0 {
+			s := strings.SplitN(t, ":", 2)
+			k := strings.TrimSpace(s[0])
+			if k == "ExceptionsList" {
+				inExceptionsList = true
+				continue
+			}
+			v := strings.TrimSpace(s[1])
+			proxyMap[k] = v
 		}
 	}
-	return ret
-}
 
-func writeConf() {
-	cfDictProxy := C.CFDictionaryRef(C.CFNetworkCopySystemProxySettings())
-	defer C.CFRelease(C.CFTypeRef(cfDictProxy))
 	darwinProxyConf = ProxyConf{}
 
-	cfNumHttpEnable := C.CFNumberRef(C.CFDictionaryGetValue(cfDictProxy, unsafe.Pointer(C.kCFNetworkProxiesHTTPEnable)))
-	if unsafe.Pointer(cfNumHttpEnable) != C.NULL && cfNumberGetGoInt(cfNumHttpEnable) > 0 {
+	// http
+	if v, ok := proxyMap["HTTPEnable"]; ok && v == "1" {
 		darwinProxyConf.Static.Active = true
 		if darwinProxyConf.Static.Protocols == nil {
 			darwinProxyConf.Static.Protocols = make(map[string]string)
 		}
-		httpHost := C.CFStringRef(C.CFDictionaryGetValue(cfDictProxy, unsafe.Pointer(C.kCFNetworkProxiesHTTPProxy)))
-		httpPort := C.CFNumberRef(C.CFDictionaryGetValue(cfDictProxy, unsafe.Pointer(C.kCFNetworkProxiesHTTPPort)))
-
-		httpProxy := fmt.Sprintf("%s:%d", cfStringGetGoString(httpHost), cfNumberGetGoInt(httpPort))
+		httpProxy := fmt.Sprintf("%s:%s", proxyMap["HTTPProxy"], proxyMap["HTTPPort"])
 		darwinProxyConf.Static.Protocols["http"] = httpProxy
 	}
 
-	cfNumHttpsEnable := C.CFNumberRef(C.CFDictionaryGetValue(cfDictProxy, unsafe.Pointer(C.kCFNetworkProxiesHTTPSEnable)))
-	if unsafe.Pointer(cfNumHttpsEnable) != C.NULL && cfNumberGetGoInt(cfNumHttpsEnable) > 0 {
+	// https
+	if v, ok := proxyMap["HTTPSEnable"]; ok && v == "1" {
 		darwinProxyConf.Static.Active = true
 		if darwinProxyConf.Static.Protocols == nil {
 			darwinProxyConf.Static.Protocols = make(map[string]string)
 		}
-		httpsHost := C.CFStringRef(C.CFDictionaryGetValue(cfDictProxy, unsafe.Pointer(C.kCFNetworkProxiesHTTPSProxy)))
-		httpsPort := C.CFNumberRef(C.CFDictionaryGetValue(cfDictProxy, unsafe.Pointer(C.kCFNetworkProxiesHTTPSPort)))
-
-		httpProxy := fmt.Sprintf("%s:%d", cfStringGetGoString(httpsHost), cfNumberGetGoInt(httpsPort))
+		httpProxy := fmt.Sprintf("%s:%s", proxyMap["HTTPSProxy"], proxyMap["HTTPSPort"])
```

Comment:
It should be:

```suggestion
		httpProxy := net.JoinHostPort(proxyMap["HTTPSProxy"], proxyMap["HTTPSPort"])
```

shouldn't it?
It may not work with IPv6 addresses.